tensordict/prototype/tree.py

```diff
@@ -0,0 +1,221 @@
+        return self._source
+
+    @source.setter
+    def _(self):
+        raise RuntimeError(
+            "Source should not be updated manually, it is managed by the nodes and "
+            "`make_tree`."
+        )
+
+    def _make_meta(self, key):
+        if key in self._children:
+            # entries of self._children are always nodes, so we can simplify slightly
+            # compared to the logic in other implementations of _make_meta
+            out = self._children[key]
+            is_memmap = self._is_memmap if self._is_memmap is not None else False
+            is_shared = (
+                self._is_shared if self._is_shared is not None else _is_shared(out)
+            )
+            return MetaTensor(
+                out,
+                device=out.device,
+                _is_memmap=is_memmap,
+                _is_shared=is_shared,
+                _is_tensordict=True,
+            )
+        return super()._make_meta(key)
+
+    def __setitem__(self, key, item):
+        key = key[0] if isinstance(key, tuple) and len(key) == 1 else key
+        if isinstance(key, tuple) or isinstance(item, TensorDictBase):
+            if isinstance(key, tuple):
+                key, subkey = key[0], key[1:]
+            else:
+                subkey = ()
+            if key in self._children:
+                node = self._children[key]
+            else:
+                node = self._source.add_node()
+                self._children[key] = node
+            if subkey:
+                node[subkey] = item
+            elif isinstance(item, TensorDictBase):
+                item_keys = set(item.keys(leaves_only=True))
+                if item_keys != self.source._leaf_keys:
+                    raise KeyError(
+                        KEY_ERR.format(leaf_keys=self.source._leaf_keys, key=item_keys)
+                    )
+                for k, v in item.items():
+                    node[k] = v
+        else:
+            if key not in self.source._leaf_keys:
+                raise KeyError(
+                    KEY_ERR.format(leaf_keys=self.source._leaf_keys, key=key)
+                )
+            super().__setitem__(key, item)
+
+    def __getitem__(self, key):
+        if isinstance(key, tuple):
+            key, subkey = key[0], key[1:]
+        else:
+            subkey = ()
+        if key in self._children:
+            if subkey:
+                return self._children[key][subkey]
+            return self._children[key]
+        if subkey:
+            raise KeyError("key not valid")
+        return super().__getitem__(key)
+
+    def get(self, key):
+        if isinstance(key, tuple):
+            key, subkey = key[0], key[1:]
+        else:
+            subkey = ()
+        if key in self._children:
+            if subkey:
+                return self._children[key].get(subkey)
+            return self._children[key]
+        if subkey:
+            raise KeyError("key not valid")
+        return super().get(key)
```

Comment:
This `get` method was a quick fix for a broken `__repr__`. I think these could be consolidated since they are very similar. In fact it's possible we could just delete `__getitem__` since I think `SubTensorDict` should delegate strings and tuples of strings to `get` anyway.